### ISSUE TITLE
modelForElement

### ISSUE THREAD

**Issue by robrez** [vaadin/vaadin-grid#1256]
### Description
Would like a `modelForElement` function or similar

### Expected outcome
```javascript
_onSomeEvent(e) {
  let model = this.$.grid.modelForElement(e.target);
}
```

### Actual outcome
```javascript
_onSomeEvent(e) {
  const path = e.composedPath();
  const cell = path.filter(item => item._instance && item.getAttribute("role") === "gridcell");
  if (cell) {
    let model = cell._instance;
  }
}
```

**Comment by moderndeveloperllc** [comment]
@robrez This might not be what you are looking for, but you can get `e.model.item` for events. It pulls in the item for that iteration. I currently do something similar to
```
_passwordReset(event) {
    let resetButton = event.path[2];
    let username = event.model.item.username;
    // Logic here
}
```

**Comment by robrez** [comment]
@moderndeveloperllc Thanks; IIRC, that does work when you add event handlers to individual templates instances:

```html
<v-grid>
  <v-grid-column>
    <template class="header">Name</template>
    <template><div on-blah="_onBlah">[[item.name]]</div></template>
  </v-grid-column>
</v-grid>
```

I was trying to have a single event listener:

```html
<v-grid on-blah="_onBlah">
  <v-grid-column>
    <template class="header">Name</template>
    <template>[[item.name]]</template>
  </v-grid-column>
</v-grid>
```

**Comment by tomivirkki** [comment]
Thanks for the suggestion @robrez Something like this would be handy.

One issue needing to be resolved is that since the default theme adds paddings to the `td` elements, there's currently no way to listen to mouse events that occur over the area covered by the paddings. Even if you add click listeners to your content elements, clicks over the padding are are ignored.

This manifests in the grid demo at https://cdn.vaadin.com/vaadin-context-menu/4.1.0-alpha1/demo/#context-menu-advanced-demos


Thinking out loud:
```
grid.addEventListener('click', e => {
 const model = grid.modelForEvent(e);
)
```
...dunno, looks hairy. In this case the click could come from anywhere inside the grid.

Perhaps an additional custom event?
```
grid.addEventListener('body-cell-click', e => {
 const model = e.detail;
)
```

**Comment by robrez** [comment]
That does look convenient. 

Fwiw, we also attach dblclick listeners to some grids. You may not want to add all of these variations of internal listeners?

**Comment by pekam** [comment]
The `detailedEvents` feature #1473 covers these use-cases, so I'm closing this issue.
The feature will be most likely shipped in version 5.3
